src/shared/user-record.c

```diff
@@ -2769,9 +2769,12 @@ bool user_name_fuzzy_match(const char *names[], size_t n_names, char **matches)
         return false;
 }
 
-int user_record_match(UserRecord *u, const UserDBMatch *match) {
+bool user_record_match(UserRecord *u, const UserDBMatch *match) {
         assert(u);
 
+        if (!uid_is_valid(u->uid))
+                return false;
```

Comment:
So, if we're going for this approach, calling this function without match should become a nop/have passthrough semantics, as mentioned above. This hence should be moved below !match check

Reply:
Swapped.